sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -544,6 +544,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val HIVE_METASTORE_PARTITION_PRUNING_FALLBACK =
+    buildConf("spark.sql.hive.metastorePartitionPruningFallback")
+      .doc("When true, enable fallback to fetch all partitions if Hive metastore partition " +
+           "push down fails. This is applicable only if partition pruning is enabled (see " +
+           s" ${HIVE_METASTORE_PARTITION_PRUNING.key}). Enabling this may degrade performance " +
+           "if there are a large number of partitions." )
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
By default, this should be false

Reply:
If we set that to false, SPARK-17992 will change the default behavior. Is that ok?

So if we set this flag to false by default, then an exception from HMS will always be re-thrown (direct sql or not). Just want to make sure that we understand that the default behavior will change.

Reply:
Done.